TESTS/netsocket/tcp/tcpsocket_echotest.cpp

```diff
@@ -71,7 +71,9 @@ void TCPSOCKET_ECHOTEST()
         fill_tx_buffer_ascii(tcp_global::tx_buffer, BUFF_SIZE);
 
         sent = sock.send(tcp_global::tx_buffer, pkt_s);
-        if (sent < 0) {
+        if (check_oversized_packets(sent, pkt_s)) {
```

Comment:
~~How about just sending the same amount of data but doing it in multiple chunks instead? We are anyway talking here about a stream instead of datagram.~~

~~If the modem returns error when trying to send too much data at a time it should also be handled.~~
Maybe instead of proactively recognizing "oversized" packets make the check if the driver returns error like NSAPI_ERROR_PARAMETER. I would say we want to have a clear indication packet is too big instead of dropping it silently, or what ever the driver might do instead.

Reply:
The check_oversized_packets() function does exactly what you said. It starts with:
` if (error == NSAPI_ERROR_PARAMETER) {`
I hid it away from the test code, as this is not a main point of any of our tests, but rather something we are forced to protect against. I think putting it all into test code would unnecessarily obfuscate it.

Reply:
Ah, sorry that I didn't comprehend what I read :)